src/test/groovy/org/shipkit/internal/gradle/AutoDiscoverGradlePluginsPluginTest.groovy

```diff
@@ -0,0 +1,48 @@
+package org.shipkit.internal.gradle
+
+import com.gradle.publish.PluginBundleExtension
+import testutil.PluginSpecification
+
+class AutoDiscoverGradlePluginsPluginTest extends PluginSpecification {
+
+    public static final String META_INF_GRADLE_PLUGINS = 'src/main/resources/META-INF.gradle-plugins'
+
+    def "Apply"() {
+        expect:
+        project.plugins.apply(AutoDiscoverGradlePluginsPlugin)
+    }
+
+    def "DiscoverGradlePluginPropertyFiles"() {
+        given:
+        project.file(META_INF_GRADLE_PLUGINS).mkdirs()
+        project.file("$META_INF_GRADLE_PLUGINS/org.shipkit.test-plugin.properties") << "implementation-class=org.shipkit.TestPlugin"
+        when:
+        project.plugins.apply(AutoDiscoverGradlePluginsPlugin)
+        project.plugins.apply("com.gradle.plugin-publish")
+        then:
+        PluginBundleExtension pluginBundleExtension = project.getExtensions().findByType(PluginBundleExtension.class);
```

Comment:
Since this is Groovy code (and a test) I suggest to use Gradle's Groovy style, it will be cleaner:

```Groovy
project.pluginBundle.plugins.size() == 1
pluginBundleExtension.plugins.testPlugin
...
```

If you prefer compile safety, keep code as is :) Keep in mind that those are tests, we execute them every time, compile time safety less important than readability.